src/box/tuple_compare.cc

```diff
@@ -906,6 +906,46 @@ key_compare(const char *key_a, hint_t key_a_hint,
 	}
 }
 
+const char min_key = '\0';
+const char max_key = '\0';
+int
+key_compare_ext(const char *key_a, hint_t key_a_hint,
+		const char *key_b, hint_t key_b_hint, struct key_def *key_def)
+{
+	assert(key_a != NULL);
+	assert(key_b != NULL);
+
+	if (key_a == key_b && (key_a == &min_key || key_a == &max_key))
+		return 0;
+
+	if (key_a == &min_key)
+		return -1;
+	if (key_a == &max_key)
+		return 1;
+	if (key_b == &min_key)
+		return 1;
+	if (key_b == &max_key)
+		return -1;
+
+	int rc = hint_cmp(key_a_hint, key_b_hint);
```

Comment:
Starting from this point, I see very similar code to the code of the key_compare function, except that in the key_compare function assert(part_count <= key_def->part_count); , but here is assert(part_count >= key_def->part_count);. Can you explain it please!